test/projectile-test.el

```diff
@@ -636,6 +636,28 @@
                          (projectile-find-matching-file
                           "spec/models/food/sea_spec.rb"))))))))
 
+(ert-deftest projectile-test-exclude-out-of-project-submodules ()
+  (projectile-test-with-files
+      ("project/.git/"
+       "project/.gitmodules"
+       "project/web-ui/"
+       "project/web-ui/.projectile"
+       "project/web-ui/vendor/client-submodule/"
+       "project/server/vendor/server-submodule/")
+    (let ((project (file-truename (expand-file-name "project/web-ui")))
+          (projectile-indexing-method 'native))
```

Comment:
Man, I checked to make sure my patch failed without my test. I'm not sure how it worked with it set to `native`.